### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2018-09-01

### PR DESCRIPTION
**Pull request by yashmuel**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/yashmuel/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2018-09-01-yashmuel-swagger-completness...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by yashmuel** [comment]
Hi @akning-ms can you please take a look on LintDiff validation issue about missing operations_API?
should i add kind of file or can get an exemption?
We don't have such a file in any of our APIs and we've an updated operational API with this API as well:
[microsoft.insight.txt](https://github.com/Azure/azure-rest-api-specs/files/5807259/microsoft.insight.txt)

**Comment by raych1** [comment]
> Hi @akning-ms can you please take a look on LintDiff validation issue about missing operations_API?
> should i add kind of file or can get an exemption?
> We don't have such a file in any of our APIs and we've an updated operational API with this API as well:
> [microsoft.insight.txt](https://github.com/Azure/azure-rest-api-specs/files/5807259/microsoft.insight.txt)

@yashmuel , it needs to add an API to expose the operations of this RP supported as ARM requirement. See [rule#R3023](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3023-operationsapiimplementation) for the rule details and fix.

**Comment by yashmuel** [comment]
@weidongxu-microsoft can you please help me merge this one as well?

**Comment by weidongxu-microsoft** [comment]
I was not following this PR.

It appears that this PR have multiple property removed, which is a big breaking changes from a previous api-version.

My suggestion is to follow up with assignee for standard review.

**Comment by yashmuel** [comment]
> I was not following this PR.
> 
> It appears that this PR have multiple property removed, which is a big breaking changes from a previous api-version.
> 
> My suggestion is to follow up with assignee for standard review.

Got it.
Just for the record:
It's an older API than the one is supposed to "break". It was our own mistake that it doesn't have a swagger.
Those APIs are just for our UI internal use and they have different response scheme and the UI guys aware of it.
It's definitely can't break anyone, as this version wasn't exposed until now and the service working the same as always.

@kairu-ms can you please help me with review and merge this?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The metric baseline Id."
+        },
+        "type": {
+          "type": "string",
+          "description": "The resource type of the metric baseline resource."
+        },
+        "metricName": {
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:163]: For api version 2019-03-01, the property name is 'name'

https://github.com/Azure/azure-rest-api-specs/blob/521ed887e283fc4de186fa59c6b02cdbcd7f73fb/specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json#L145-L147

Comment (kairu-ms) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:163]: Please make sure you want to rename it.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+          },
+          "description": "The baseline values for each sensitivity."
+        },
+        "metadataValues": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BaselineMetadata"
+          },
+          "description": "The baseline metadata values."
+        }
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:218]: Different from 2019-03-01, Please make sure you want to rename it.
https://github.com/Azure/azure-rest-api-specs/blob/521ed887e283fc4de186fa59c6b02cdbcd7f73fb/specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json#L223-L229

**Comment by yashmuel** [comment]
@kairu-ms i think our source code might help you review.
here is 2018 response DTO:
image
and here is 2019 response DTO:
image

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json (hunk 1 above)
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:163]: checked.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json (hunk 2 above)
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:218]: checked. i'm sure.

**Comment by yashmuel** [comment]
@kairu-ms verified your comments. thanks!
if you want to be sure, here you can find a response example of our 2018-09-01 api version:
https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=7099E989-4EC7-4296-96FC-D7A90C3348C8

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json (hunk 2 above)
Comment (yashmuel) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2018-09-01/metricBaselines_API.json:218]: @asafst do you think we should rename 2019-03-01 from metadata to metadataValues as well? we're not flagged on this, but i think we could be.

**Comment by yashmuel** [comment]
@asafst i think i didn't send you this one for review, please take a look.

**Review by asafst — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**